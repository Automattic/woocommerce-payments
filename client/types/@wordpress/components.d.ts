/**
 * External dependencies
 */
import '@wordpress/components';

/**
 * External dependencies declarations
 */

// Can be removed after updating the package to a newer version.
declare module '@wordpress/components' {
	namespace DropdownMenu {
		interface BaseProps {
			/**
			 * Text to display on the nested `Button` component in the `renderToggle`
			 * implementation of the `Dropdown` component used internally.
			 */
			text?: string | JSX.Element;
		}
	}
}
