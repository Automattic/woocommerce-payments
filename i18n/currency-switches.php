<?php
/**
 * Currency changes information
 *
 * @package WooCommerce\Payments\i18n
 * @version 5.2.0
 */

defined( 'ABSPATH' ) || exit;

return [
	'HR' => [
		'effective' => 1672531200, // Jan 1, 2023.
		'update'    => [
			'currency_code' => 'EUR',
			'name'          => 'Euro',
			'singular'      => 'euro',
			'plural'        => 'euros',
			'short_symbol'  => 'â‚¬',
			'locales'       => $locales['EUR'],
		],
	],
];
