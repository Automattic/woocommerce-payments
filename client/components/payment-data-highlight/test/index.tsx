/** @format */
/**
 * External dependencies
 */
import React from 'react';
import { __ } from '@wordpress/i18n';

/**
 * External dependencies
 */
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import PaymentDataHighlight from '../';

describe( 'PaymentDataHighlight', () => {
	test( 'renders PaymentDataHighlight with tooltip', () => {
		const { container: data } = render(
			<PaymentDataHighlight
				label="Deposits"
				amount={ '€3,143.00' }
				change={ 22 }
				reportUrl="#"
				tooltip={ __(
					"A charge is the amount billed to your customer's payment method.",
					'woocommerce-payments'
				) }
			/>
		);

		expect( data ).toMatchSnapshot();
	} );

	test( 'renders PaymentDataHighlight without tooltip', () => {
		const { container: data } = render(
			<PaymentDataHighlight
				label="Transactions"
				amount={ '€2,143.00' }
				change={ 22 }
				reportUrl="#"
			/>
		);

		expect( data ).toMatchSnapshot();
	} );
} );
