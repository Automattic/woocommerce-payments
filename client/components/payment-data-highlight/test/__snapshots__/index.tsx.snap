// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentDataHighlight renders PaymentDataHighlight with tooltip 1`] = `
<div>
  <div
    class="payment-data-highlight__wrapper"
  >
    <div
      class="payment-data-highlight__visible"
    >
      <div
        class="payment-data-highlight__label"
      >
        <span>
          Deposits
        </span>
        <button
          class="wcpay-tooltip__content-wrapper wcpay-tooltip--click__content-wrapper"
          type="button"
        >
          <div
            class="wcpay-tooltip__content-wrapper"
          >
            <div
              role="button"
              tabindex="0"
            >
              <svg
                class="gridicon gridicons-help-outline"
                height="16"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M12 4c4.411 0 8 3.589 8 8s-3.589 8-8 8-8-3.589-8-8 3.589-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm4 8a4 4 0 00-8 0h2c0-1.103.897-2 2-2s2 .897 2 2-.897 2-2 2a1 1 0 00-1 1v2h2v-1.141A3.991 3.991 0 0016 10zm-3 6h-2v2h2v-2z"
                  />
                </g>
              </svg>
            </div>
          </div>
        </button>
      </div>
      <div
        class="payment-data-highlight__amount"
      >
        <div
          class="payment-data-highlight__amount-number"
        >
          €3,143.00
        </div>
        <div
          class="payment-data-highlight__amount-change"
        >
          <div
            class="payment-data-highlight__amount-change-arrow-up"
          >
            <span
              class="payment-data-highlight__amount-change-icon"
            >
              <svg
                class="gridicon gridicons-arrow-up"
                height="12"
                viewBox="0 0 24 24"
                width="12"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M13 20V7.83l5.59 5.59L20 12l-8-8-8 8 1.41 1.41L11 7.83V20h2z"
                  />
                </g>
              </svg>
            </span>
            <span
              class="payment-data-highlight__amount-change-arrow-up-percentage"
            >
              22
              %
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="payment-data-highlight__view-report"
    >
      <a
        data-link-type="wc-admin"
        href="#"
      >
        View report
      </a>
    </div>
  </div>
</div>
`;

exports[`PaymentDataHighlight renders PaymentDataHighlight without tooltip 1`] = `
<div>
  <div
    class="payment-data-highlight__wrapper"
  >
    <div
      class="payment-data-highlight__visible"
    >
      <div
        class="payment-data-highlight__label"
      >
        <span>
          Transactions
        </span>
      </div>
      <div
        class="payment-data-highlight__amount"
      >
        <div
          class="payment-data-highlight__amount-number"
        >
          €2,143.00
        </div>
        <div
          class="payment-data-highlight__amount-change"
        >
          <div
            class="payment-data-highlight__amount-change-arrow-up"
          >
            <span
              class="payment-data-highlight__amount-change-icon"
            >
              <svg
                class="gridicon gridicons-arrow-up"
                height="12"
                viewBox="0 0 24 24"
                width="12"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M13 20V7.83l5.59 5.59L20 12l-8-8-8 8 1.41 1.41L11 7.83V20h2z"
                  />
                </g>
              </svg>
            </span>
            <span
              class="payment-data-highlight__amount-change-arrow-up-percentage"
            >
              22
              %
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="payment-data-highlight__view-report"
    >
      <a
        data-link-type="wc-admin"
        href="#"
      >
        View report
      </a>
    </div>
  </div>
</div>
`;
